parlai/tasks/blended_skill_talk/worlds.py

```diff
@@ -0,0 +1,123 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Facebook, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+import json
+import random
+import os
+from copy import deepcopy
+
+from parlai.core.worlds import DialogPartnerWorld, validate
+from parlai.tasks.self_chat.worlds import SelfChatWorld as SelfChatBaseWorld
+
+
+def load_personas(opt):
+    print('[ loading personas.. ]')
+    fname = os.path.join(
+        opt['datapath'], 'blended_skill_talk', opt['datatype'] + '.json'
+    )
+    with open(fname) as json_file:
+        data = json.load(json_file)
+    contexts = []
+    for d in data:
+        p1 = (
+            'your persona: '
+            + d['personas'][0][0]
+            + '\n'
+            + 'your persona: '
+            + d['personas'][0][1]
+            + '\n'
+        )
+        p2 = (
+            'your persona: '
+            + d['personas'][1][0]
+            + '\n'
+            + 'your persona: '
+            + d['personas'][1][1]
+            + '\n'
+        )
+        if d['context_dataset'] == 'wizard_of_wikipedia':
+            p1 += d['additional_context'] + '\n'
```

Comment:
i think this will lead to same problem @jaseweston just had where you are putting an extra newline at the end of the context. are you sure you want that here?

Reply:
Hmm - in what cases will the end of the context will have an extra newline? A few lines down, `p1` and `p2` will get `ctxt` added to them, and `d['free_turker_utterance']` and `d['guided_turker_utterance']` will both always be non-empty